### PR TITLE
(Refactor)[Statistics] Fix lock risks in Statistics Job

### PR DESCRIPTION
**Pull request by EmmyMiao87**
# Proposed changes

1. Remove lock nesting between job and task
2. Solve the deadlock problem during job update
3. Avoid printing the log while holding the lock

WIP by @weizhengte 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJobScheduler.java [hunk 1]
```diff
@@ -86,7 +86,11 @@ protected void runAfterCatalogReady() {
                 LOG.info("The statistics task queue is full, schedule the job(id={}) later", pendingJob.getId());
             } catch (DdlException e) {
                 pendingJobQueue.remove();
-                pendingJob.updateJobState(StatisticsJob.JobState.FAILED);
+                try {
+                    // TODO change to without exception
+                    pendingJob.updateJobState(StatisticsJob.JobState.FAILED);
+                } catch (DdlException ddlException) {
+                }
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJobScheduler.java:93]: catch but no action ?

Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJobScheduler.java:93]: I will add a LOG(FATAL)

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**